cli/cmd/org/rename.go

```diff
@@ -26,65 +24,71 @@ func RenameCmd(ch *cmdutil.Helper) *cobra.Command {
 				return err
 			}
 
-			fmt.Println("Warn: Renaming an org would invalidate dashboard URLs")
-
-			if !cmd.Flags().Changed("org") && ch.Interactive {
-				orgNames, err := OrgNames(ctx, ch)
-				if err != nil {
-					return err
+			resp, err := client.GetOrganization(ctx, &adminv1.GetOrganizationRequest{Name: name})
+			if err != nil {
+				if st, ok := status.FromError(err); ok && st.Code() == codes.NotFound {
+					ch.PrintfError("org %q doesn't exist, run 'rill org list' to see available orgs", name)
+					return nil
 				}
+				return err
+			}
+			org := resp.Organization
 
-				name, err = cmdutil.SelectPrompt("Select org to rename", orgNames, "")
-				if err != nil {
-					return err
-				}
+			// Build update request
+			req := &adminv1.UpdateOrganizationRequest{Name: org.Name}
+
+			if req.NewName != nil && req.DisplayName == nil && org.DisplayName != "" {
+				ch.PrintfWarn("\nWarn: Changing org name will invalidate dashboard URLs.\n")
 			}
```

Comment:
The warning here doesn't match the `if` check. And conversely, the warning printed here was previously accompanied by a confirmation prompt. I think you want this here:

```suggestion
			if req.NewName != nil {
				ch.PrintfWarn("Changing the name will invalidate dashboard URLs.\n")
				ok, err := cmdutil.ConfirmPrompt("Do you want to continue?", "", false)
				if err != nil {
					return err
				}
				if !ok {
					ch.PrintfWarn("Aborted\n")
					return nil
				}
			}
```

And then the note about having forgotten to update the display name should go after the update (since it can easily be done as a follow up). See my next comment.